src/common/lib/transport/connectionmanager.ts

```diff
@@ -364,8 +364,20 @@ class ConnectionManager extends EventEmitter {
             'reattempting connection'
           );
           this.requestState({ state: 'connecting' });
+        } else if (this.state == this.states.connecting) {
+          // RTN20c: if 'online' event recieved while CONNECTING, abandon connection attempt and retry
+          this.pendingTransports.forEach(function (transport) {
```

Comment:
can just call `this.disconnectAllTransports()`, no need to reimplement

(in particular -- there's a subtle bug here cause forEach isn't safe when the callback mutates the array being iterated over, which `disconnectAllTransports()` avoids by using `Utils.safeArrForEach`)

Reply:
i did try `disconnectAllTransports` in my initial attempt at this but that caused connection state side effects, i will check again if there's any easy way around that but otherwise will just move to safeArrForEach

Reply:
OK I've changed[f22e9af](https://github.com/ably/ably-js/pull/1171/commits/f22e9af9b82f21b5c05982b867c8d4eb02353f24) to still use `forEach` to call `.off` on all pending transports but now it uses `disconnectAllTransports` to clean up cm state